tools/sgmdformat/tools.go

```diff
@@ -60,3 +63,26 @@ func PrepareCommand(ctx context.Context) error {
 	}
 	return nil
 }
+
+// list markdown files known by git + untracked ones.
+func listMarkdownFiles(ctx context.Context) []string {
+	output := strings.TrimSpace(
+		sg.Output(sggit.Command(
+			ctx,
+			"ls-files",
+			"*.md",               // only markdown files
+			"--others",           // include untracked files
+			"--exclude-standard", // exclude ignored files
+			"--cached",           // include "normal" files
+		)),
+	)
+	if len(output) == 0 {
+		return nil
+	}
+	files := strings.Split(output, "\n")
+	for i, file := range files {
+		// Convert to absolute path
+		files[i] = sg.FromGitRoot(file)
```

Comment:
Will this call git for every single file? Could this be a performance concern?

Reply:
On my machine with 5 md files to format, this step takes 4-5ms so quite fast. That said though it is not needed and was more of a "hängslen and livrem" kinda thing. 

I will remove it :)

Reply:
It was a great idea. We can introduce it again if we run into issues.